### PR TITLE
Remove USubscription CreateTopic() and DeprecateTopic() APIs

### PR DESCRIPTION
**Pull request by stevenhartley**
The topic lifecycle APIs were added as an optimization early on so that subscribers could know if the topic they are subscribing to is valid and the producer is alive however this optimization has given us nothing but heartburn in production as it forces all publishers to call the API when then are initialized and it causes race conditions on initial boot-up. Furthermore, validation of topics could already be done by calling UDiscovery so there was no added benefit of adding the extra states to the subscription service.

#96

### CODE REVIEW

File: up-l3/usubscription/v3/README.adoc [hunk 1]
```diff
@@ -142,53 +106,26 @@ Subscribers are automatically registered to receive subscription change notifica
 
 === Publishers
 
-* *MUST* *NOT* be permitted to call `RegisterForNotifications()` for topics they do not create through the  `CreateTopic()`, only the topics they produce
+* *MUST* *NOT* be permitted to call `RegisterForNotifications()` for topics they do not produce (i.e. `UEntity` of the producer is not equal to the `UEntity` in the passed `UUri`)
```
Comment (sophokles73) [inline up-l3/usubscription/v3/README.adoc:109]:
```suggestion
* *MUST* *NOT* be permitted to call `RegisterForNotifications()` for topics they do not produce, i.e. the uEntity sending the request is not equal to the `UEntity` of the request message's `UUri` parameter.
```

File: up-l3/usubscription/v3/README.adoc [hunk 2]
```diff
@@ -142,53 +106,26 @@ Subscribers are automatically registered to receive subscription change notifica
...
+Dispatchers are also permitted to register for subscription change notifications to facilitate the multicasting of messages. The mechanics and requirements of the dispatcher and uSubscription communication are platform deployment specific. 
 
 == Event Delivery
 
-uSubscription service, in addition to managing subscriptions for subscribers to topics, also plays a key role in providing event delivery configuration details as not every platform delivers events using the same mechanism. 
-
-the `EventDeliveryConfig` message is one of the fields returned to the subscriber to provide details of how to consume the data (how events are delivered). For example if the subscriber has to consume from a different topic or from different event infrastructure, this message will store those delivery semantics.
+uSubscription service, also provides a means to notify subscribers of how they shall consume the published events though the  `EventDeliveryConfig` returned in the `SubscriptionResponse` message. For example if the subscriber has to consume from a different topic or from different message infrastructure, this message will store those delivery semantics.
```
Comment (sophokles73) [inline up-l3/usubscription/v3/README.adoc:118]:
```suggestion
The uSubscription service also allows to indicate to subscribers how they shall consume published events by means of the `EventDeliveryConfig` returned in the `SubscriptionResponse` message. For example, if the subscriber has to consume from a different topic or from different messaging infrastructure, this message will store the corresponding delivery semantics.
```

File: up-l3/usubscription/v3/README.adoc [hunk 3]
```diff
@@ -142,53 +106,26 @@ Subscribers are automatically registered to receive subscription change notifica
...
 
 NOTE: Delivery semantics (if any) are deployment specific and not covered in this specification
 
 == uSubscription Sequences
 
-In the following section, we will elaborate on the various subscription flows for local and remote topics. When a consumer subscribes to a remote topic, it is the responsibility of the Subscription Service to relay the subscription request to the remote Subscription Service as can be illustrated in the sequence diagrams below.
-
-For the flows below we will use a fictitious device called Device1 hosting a service called uExample that has a resource that emits a Message.
+In the following section, we will elaborate on the various subscription flows for local and remote topics. When a consumer subscribes to a remote topic, it is the responsibility of the Subscription Service to relay the subscription request to the remote Subscription Service as can be seen in the sequence diagrams below.
```
Comment (sophokles73) [inline up-l3/usubscription/v3/README.adoc:124]:
```suggestion
In the following section, we will elaborate on the various subscription flows for local and remote topics. When a consumer subscribes to a remote topic, it is the responsibility of the (local) uSubscription service to relay the subscription request to the remote uSubscription service as can be seen in the sequence diagrams below.
```

**Review by sophokles73 — APPROVED** [review]
LGTM

**Comment by sophokles73** [comment]
@AnotherDaniel this might be of interest to you as well, since you are working on an implementation ...

**Comment by lynn2009** [comment]
@stevenhartley , The uSubscription still requires verification of whether the requested topic is valid. By removing the createTopic method, does this imply that uSubscription is depending on uDiscovery for functionality?

**Comment by stevenhartley** [comment]
> @stevenhartley , The uSubscription still requires verification of whether the requested topic is valid. By removing the createTopic method, this implies a strong dependency on uDiscovery for its functionality.

@lynn2009 , why do you need to check the topic is valid or not?

**Comment by lynn2009** [comment]
If uSubscription does not check the topic is valid or not, we will run into the case the subscriber can subscribe an invalid topic and get success response.

**Comment by stevenhartley** [comment]
> If uSubscription does not check the topic is valid or not, we will run into the case the subscriber can subscribe an invalid topic and get success response.

correct

**Comment by mayureshkar** [comment]
@stevenhartley  @lynn2009  with createTopic removed, what constitutes an 'invalid topic'?

**Comment by stevenhartley** [comment]
invalid UUri

**Pull request merged**